### PR TITLE
fix: build error due to survey status becoming optional

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
The web build was failing because of the display status now being an optional field. This PR fixes it by
1. we just needed the displays count in the weekly summary so instead of the status, we now do that through ID field
2. In the lib/surveyState, the TResponseUpdate will not have a displayId field now so removed that as well from this constructor
3. Removes commented method

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Review by mattinannt — APPROVED** [review]

File: apps/web/app/api/cron/weekly_summary/route.ts [hunk 1]
```diff
@@ -180,7 +180,7 @@ const getProducts = async (): Promise<ProductData[]> => {
                   },
                 },
                 select: {
-                  status: true,
+                  responseId: true,
```
Comment (mattinannt) [inline apps/web/app/api/cron/weekly_summary/route.ts:183]: @ShubhamPalriwala only the select statement changed here. So is the value even used? Or are there more changes needed?

Comment (mattinannt) [reply apps/web/app/api/cron/weekly_summary/route.ts:183]: Ahh, ok, think I got it :-)

Comment (mattinannt) [reply apps/web/app/api/cron/weekly_summary/route.ts:183]: I think it still breaks for existing Displays that don't have the responseId and is not able to calculate the response rate correctly, right?

Comment (ShubhamPalriwala) [reply apps/web/app/api/cron/weekly_summary/route.ts]: i've discussed with @Dhruwang and updated this field with the always-present `id` field. Please take a look now

**Pull request merged**